pep-9999.rst

```diff
@@ -304,10 +304,104 @@ Security Implications
+   package ecosystem has a corresponding mapping. If no appropriate PURL exists,
+   the package maintainer may consider submitting a new PURL to the central registry.
+
+An overall workflow diagram might look like this:
+
+.. mermaid::
+
+   flowchart TD
+      A[Python package author with new external dependency] --> |Looks in| B(PURL/description collection)
+      B --> | Find PURL OK | E(Add PURL to pyproject.toml)
+      A --> | Looks in | C(Ecosystem mapping file)
+      C --> | Finds familiar ecosystem package name | D(Inverse map ecosystem package name to PURL)
+      D --> | Mapping exists | E
+      B --> | PURL not found | F(Submit PURL proposal to PURL/description collection)
+      F --> | Accepted | G(Mapping maintainers notified of missing PURL mappings)
+      D --> | Mapping missing. User looks in PURL collection. | B
+      B --> | Was mapping missing? | H(User may contribute entry to mapping)
+
+Package ecosystem maintainers usage
+-----------------------------------
+
+Any packages that express a PURL dependency that does not have a mapping in a given package
+ecosystem will mean a failed installation. It is thus critically important that each package
+ecosystem maintain their mappings. Key to this will be automation. Some ideas for automation
+are:
+
+1. Alert mapping maintainers whenever a new PURL is added to the registry (probably noisy)
+2. Provide tools that allow maintainers to diff their mappings to the registry contents to
+   quickly identify missing entries
+3. Provide automated tooling that submits PRs to known mapping locations, such that maintainers
+   need only fill in the ecosystem package name
+4. Provide status for each PURL, to readily identify which PURLs need attention
+
+This maintenance is likely to be a lot of work to establish the initial mapping, but ideally small
+on an ongoing basis.
+
+
+Central PURL registry maintainers
+---------------------------------
+
+Central PURL registry maintainers curate the collection of PURLs. These contributors
+need to be able to refer to clearly defined rules for when a new PURL can be defined. It is
+undesirable to be loose with PURL definition, because each definition implies maintenance in
+the mappings in many other places.
+
+
+End user package consumers
+--------------------------
+
+In an ideal case, these package install commands can be done transparently, and the user 
+experience is the same as the self-contained binary wheels of today. There are several
+foreseeable issues that will arise, though:
+
+* A mapping does not exist for the user's desired package ecosystem
+* A user does not have permissions to run the install commands provided by our 
+  tool (e.g. system python users)
+
+If packagers suddenly drop self-contained wheels before the mapping can be well-established,
+there will likely be a period of negative user experience.
+
+TODO: can we set a target adoption level before we think it's safe to switch?
```

Comment:
How do we measure this?

Reply:
I was thinking fractional download counts. Like for a given project, how many of the downloads of the latest version are the self-contained wheels vs. the shared-library variety?

Reply:
If we can track both separately, that works, yes.